cmake/kokkos_functions.cmake

```diff
@@ -799,3 +799,36 @@ ENDFUNCTION(COMPILER_SPECIFIC_DEFS)
 FUNCTION(COMPILER_SPECIFIC_LIBS)
   COMPILER_SPECIFIC_OPTIONS_HELPER(${ARGN} LINK_LIBRARIES)
 ENDFUNCTION(COMPILER_SPECIFIC_LIBS)
+
+# Given a list of the form
+#  key1;value1;key2;value2,...
+# Create a list of all keys in a variable named ${KEY_LIST_NAME}
+# and set the value for each key in a variable ${VAR_PREFIX}key1,...
+# kokkos_key_value_map(ARCH ALL_ARCHES key1;value1;key2;value2)
+# would produce a list variable ALL_ARCHES=key1;key2
+# and individual variables ARCHkey1=value1 and ARCHkey2=value2
+MACRO(kokkos_key_value_map VAR_PREFIX KEY_LIST_NAME)
+  SET(PARSE_KEY ON)
+  SET(${KEY_LIST_NAME})
+  FOREACH(ENTRY ${ARGN})
+    IF(PARSE_KEY)
+      SET(CURRENT_KEY ${ENTRY})
+      SET(PARSE_KEY OFF)
+      LIST(APPEND ${KEY_LIST_NAME} ${CURRENT_KEY})
+    ELSE()
+      SET(${VAR_PREFIX}${CURRENT_KEY} ${ENTRY})
+      SET(PARSE_KEY ON)
+    ENDIF()
+  ENDFOREACH()
+ENDMACRO()
+
+FUNCTION(kokkos_check_deprecated_options)
+  kokkos_key_value_map(DEPRECATED_MSG_ DEPRECATED_LIST ${ARGN})
```

Comment:
function name should be uppercase for consistency

Reply:
@jjwilke Do you want to fix that or merge as is?

Reply:
I can fix it.